### DISCUSSION TITLE
Xterm.Js not work in with NextJs | React app work fine

### DISCUSSION THREAD

**Discussion by MwSpaceLLC** [vercel/next.js#22409] [Help] [answered]
**What version of Next.js are you using?**

10.0.7

**What version of Node.js are you using?**

v12.16.3

**What browser are you using?**

chrome

**What operating system are you using?**

Windows

**How are you deploying your application?**

next dev

**Describe the Bug**

Hi there,
We want work with this js library, but import and require not work property, in a normal react application, this code works without problems. But in next.js we fail to use this library

>NEXTJS >>https://codesandbox.io/s/nervous-http-jv3ip?file=/pages/index.js

![image](https://user-images.githubusercontent.com/29952045/108597898-9e650100-738b-11eb-8d5d-a224cd89288b.png)

>REACTJS>>https://codesandbox.io/s/exciting-noyce-1qz7u?file=/src/App.js

![image](https://user-images.githubusercontent.com/29952045/108597713-c9028a00-738a-11eb-9ddd-040826ecbed4.png)

This library is very important to us because we need to create a container manager with WebSocket.

**Comment by MwSpaceLLC** [comment, marked as answer]
Hi guys, for all those who are having these problems we have solved by reading the NextJs guide.

If we have to work with components that require access to JavaScript properties such as Window or localstorage, we need to import these dynamically, disabling the Side rendering server.

>> https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr

![image](https://user-images.githubusercontent.com/29952045/108702918-31c44080-750a-11eb-94bf-596fe89a1311.png)

In this case, The error occurs because **xterm** tries to access the **window object** which is **not available on the server-side**. To fix it, you can dynamically import **xterm** with ssr: false so it only gets loaded on the client-side.

![image](https://user-images.githubusercontent.com/29952045/108703058-6506cf80-750a-11eb-9113-be812c8b2041.png)

>> https://codesandbox.io/s/adoring-wright-64nro?file=/pages/index.js

**Reply by ndungtse** [reply]
Solved my problem!! Thanks 😍
